### PR TITLE
VNet/Vxlan HLD

### PR DESCRIPTION
**Pull request by prsunny**
High level design for Sonic Vxlan

### CODE REVIEW

File: doc/vxlan/Vxlan_hld.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+
+```
+VXLAN_TUNNEL:{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}}
+
+VXLAN_TUNNEL_MAP|{{tunnel_name}}|{{tunnel_map}}
+    "vni_id": {{ vni_id}}
+    "vlan_id": {{ vlan_id }}
```
Comment (pavel-shirshov) [inline doc/vxlan/Vxlan_hld.md]: It's already supported as `vni` and `vlan`. Check https://github.com/Azure/sonic-swss/blob/master/orchagent/vxlanorch.h#L56

File: doc/vxlan/Vxlan_hld.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+```
+### 2.1.2 VNET Table
+```
+VNET:{{vnet_name}} 
+    "vxlan_tunnel": {{tunnel_name}}
+    "vxlanid": {{vni}} 
+    "peer_list": {{vnet_name_list}}
```
Comment (pavel-shirshov) [inline doc/vxlan/Vxlan_hld.md]: Probable it's better to have one name of the attribute to refer to vni? I'd put `vni` here.

File: doc/vxlan/Vxlan_hld.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+VNI_ID                                = DIGITS                        ; 1 to 16 million VNI values
+PEER_LIST                             = \*vnet_name                   ; vnet names seperate by "," 
+                                                                             (empty indicates no peering)
+```
+
+```
+; Defines schema for VNet Interface configuration attributes
+key                                   = VNET_INTF:name:prefix         ; Vnet interface name with IP prefix
+; field                               = value
+VNET_NAME                             = vnet_name                     ; vnet name where the interface belongs to
```
Comment (jipanyang) [inline doc/vxlan/Vxlan_hld.md]: For the schemas defined here,  are they for appDB or configDB?

**Comment by lguohan** [comment]
for vxlan decap flow, how you decide which pipeline the packet take after decap, l2 or l3?

File: doc/vxlan/Vxlan_hld.md (hunk 3 above)
Comment (lguohan) [reply doc/vxlan/Vxlan_hld.md]: better to specify this clearly.

File: doc/vxlan/Vxlan_hld.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    "vnet_name": {{vnet_name}} 
+```
+
+### 2.1.2 Schemas
+```
+; Defines schema for VXLAN Tunnel configuration attributes
+key                                   = VXLAN_TUNNEL:name             ; Vxlan tunnel configuration
+; field                               = value
+SRC_IP                                = ipv4                          ; Ipv4 source address, lpbk address
+DST_IP                                = ipv4                          ; Ipv4 source address, lpbk address
```
Comment (stcheng) [inline doc/vxlan/Vxlan_hld.md]: the comment should be the destination address

File: doc/vxlan/Vxlan_hld.md (hunk 3 above)
Comment (MELLANOX-zhichenr) [reply doc/vxlan/Vxlan_hld.md]: I noticed that VNI to VLAN map is added to config DB, while VNI to virtual router map is added to app DB.
Is there a reason for separating these two types of tunnel maps?

Comment (prsunny) [reply doc/vxlan/Vxlan_hld.md]: Yes, the idea is to have VxlanOrch only deal with Tunnel configurations (Encap/Decap etc). The peering must be required only by VRF/RouterOrch.

File: doc/vxlan/Vxlan_hld.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+	- config vxlan <vxlan_name> vlan <vlan_id> flood vtep <ip1, ip2, ip3>
+	- show mac vxlan <vxlan_name> <vni_id>
+	- show vxlan <vxlan_name>
+```
+Configuring VNet peering via CLI is beyond the scope
+
+## 1.4 Scalability requirements
+
+### 2.1.1 VNet Peering
+###### Table 2: VNet peering scalability
```
Comment (tushar-ty) [inline doc/vxlan/Vxlan_hld.md]: What will be the total number of routes to be supported ? Can that be added to this table as well. - thx

**Pull request merged**